luigi/scheduler.py

```diff
@@ -287,14 +327,28 @@ class SimpleTaskState(object):
     def __init__(self, state_path):
         self._state_path = state_path
         self._tasks = {}  # map from id to a Task object
+        self._batch_tasks = {}  # map from family to TaskBatch object
+        self._running_batches = {}  # map from id to list of batched task ids
         self._status_tasks = collections.defaultdict(dict)
         self._active_workers = {}  # map from id to a Worker object
 
     def get_state(self):
-        return self._tasks, self._active_workers
+        return {
+            TASKS: self._tasks,
+            ACTIVE_WORKERS: self._active_workers,
+            BATCH_TASKS: self._batch_tasks,
+            RUNNING_BATCHES: self._running_batches,
+        }
 
     def set_state(self, state):
-        self._tasks, self._active_workers = state
+        if isinstance(state, dict):
+            self._tasks = state.get(TASKS, {})
+            self._active_workers = state.get(ACTIVE_WORKERS, {})
+            self._batch_tasks = state.get(BATCH_TASKS, {})
+            self._running_batches = state.get(RUNNING_BATCHES, {})
+        else:
+            self._tasks, self._active_workers = state
```

Comment:
Can you add a note that this is only for backward compatibility and this codepath will be removed?